src/Sentry/PlatformAbstractions/NetFxInstallationsEventProcessor.cs

```diff
@@ -0,0 +1,68 @@
+#if NETFX
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Sentry.Extensibility;
+using Sentry.PlatformAbstractions;
+
+namespace Sentry.PlatformAbstractions
+{
+    internal class NetFxInstallationsEventProcessor : ISentryEventProcessor
+    {
+        internal static readonly string NetFxInstallationsKey = ".NET Framework";
+
+        private readonly Lazy<Dictionary<string, string>> _netFxInstallations =
+            new Lazy<Dictionary<string, string>>(() => GetInstallationsDictionary(), LazyThreadSafetyMode.ExecutionAndPublication);
+
+        private bool _netFxInstallationEnabled = true;
+
+        private readonly SentryOptions _options;
+
+        internal NetFxInstallationsEventProcessor(SentryOptions options)
+        {
+            Debug.Assert(options != null);
+            _options = options;
+        }
+
+        internal static Dictionary<string, string> GetInstallationsDictionary()
+        {
+            var versionsDictionary = new Dictionary<string, string>();
+            var installations = FrameworkInfo.GetInstallations();
+            foreach (var profile in installations.Select(p => p.Profile).Distinct())
```

Comment:
Why `Distinct` here? Isn't this going to work on the object references?

Reply:
The goal of distinct is to avoid repeating keys to be inserted into the Dictionary and the foreach loop will then group the releases by profiles.

Reply:
But you're doing `Distinct` on a reference type so unless `GetHashCode` and `Equals` was overriden, it will dedupe on the reference.

Reply:
Confirmed offline that it fails with dupe key in Map otherwise